<?xml version="1.0"?>
<launch>
    <arg name="base_link_frame" default="base_link"/>
    <arg name="map_frame" default="map"/>
    <arg name="twist_topic" default="/twist"/>
    <arg name="pose_topic" default="/pose"/>
    <arg name="estimate_3d_pose" default="false"/>
    <arg name="update_rate" default="30"/>
    <arg name="num_particles" default="100"/>
    <arg name="publish_frame" default="false"/>
    <arg name="publish_marker" default="true"/>
    <arg name="respawn" default="false"/>
    <arg name="expansion_reset_ess_threashold" default="30"/>
    <arg name="max_expantion_orientation" default="0.1"/>
    <arg name="max_expantion_position" default="1.0"/>
    <arg name="sensor_reset_ess_threashold" default="10"/>
    <arg name="max_sensor_reset_orientation" default="0.5"/>
    <arg name="max_sensor_reset_position" default="1.0"/>
    <arg name="sensor_reset_radius" default="5.0"/>
    <arg name="weight_position" default="0.5"/>
    <arg name="weight_orientation" default="0.5"/>

    <node pkg="pf_localization" type="pf_localization_node" name="pf_localization_node" output="screen" respawn="$(arg respawn)" respawn_delay="0">
        <param name="base_link_frame" value="$(arg base_link_frame)"/>
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="pose_topic" value="$(arg pose_topic)"/>
        <param name="twist_topic" value="$(arg twist_topic)"/>
        <param name="estimate_3d_pose" value="$(arg estimate_3d_pose)"/>
        <param name="update_rate" value="$(arg update_rate)"/>
        <param name="num_particles" value="$(arg num_particles)"/>
        <param name="publish_marker" value="$(arg publish_marker)"/>
        <param name="publish_frame" value="$(arg publish_frame)"/>
        <param name="expansion_reset_ess_threashold" value="$(arg expansion_reset_ess_threashold)"/>
        <param name="max_expantion_orientation" value="$(arg max_expantion_orientation)"/>
        <param name="max_expantion_position" value="$(arg max_expantion_position)"/>
        <param name="sensor_reset_ess_threashold" value="$(arg sensor_reset_ess_threashold)"/>
        <param name="max_sensor_reset_orientation" value="$(arg max_sensor_reset_orientation)"/>
        <param name="max_sensor_reset_position" value="$(arg max_sensor_reset_position)"/>
        <param name="sensor_reset_radius" value="$(arg sensor_reset_radius)"/>
        <param name="weight_orientation" value="$(arg weight_orientation)"/>
        <param name="weight_position" value="$(arg weight_position)"/>
    </node>
</launch>